--- pie/tests/top_down.rs
+++ pie/tests/top_down.rs
@@ -131,9 +131,7 @@
   // 2) Require `ToLower` again and assert that no tasks are executed because all dependencies are consistent:
   // → ToLower
   //   ? ReadFile
-  //     → ReadFile
-  //       ✓ `file`
-  //     ← Ok(String("HELLO WORLD!"))
+  //     ✓ `file`
   //   ✓ ReadFile
   // ← Ok(String("hello world!"))
   // 🏁
@@ -146,12 +144,10 @@
   // 3) Require `ToLower` and assert that both tasks are re-executed in reverse dependency order:
   // → ToLower
   //   ? ReadFile
-  //     → ReadFile
-  //       ✗ `file` [inconsistent: modified file stamp change]
-  //       ▶ ReadFile [reason: `file` is inconsistent due to modified file stamp change]
-  //         - `file`
-  //       ◀ Ok(String("!DLROW OLLEH")) [note: returns a different output!]
-  //     ← Ok(String("!DLROW OLLEH"))
+  //     ✗ `file` [inconsistent: modified file stamp change]
+  //     ▶ ReadFile [reason: `file` is inconsistent due to modified file stamp change]
+  //       - `file`
+  //     ◀ Ok(String("!DLROW OLLEH")) [note: returns a different output!]
   //   ✗ ReadFile [inconsistent: equals output stamp change]
   //   ▶ ToLower [reason: ReadFile is inconsistent due to equals output stamp change]
   //     → ReadFile
@@ -200,10 +196,7 @@
   // Require and execute ends come after require and execute starts.
   assert!(require.end() > require.start());
   assert!(execute.end() > execute.start());
-  // A task is only executed if it is required.
-  // - Task execute starts should be later than their requires.
-  assert!(execute.start() > require.start());
-  // - Task require ends should be later than their executes.
+  // Task require ends should be later than their executes.
   assert!(require.end() > execute.end());
 }
 
