--- pie/src/stamp.rs
+++ pie/src/stamp.rs
@@ -2,10 +2,10 @@
 
 #[cfg(test)]
 mod test {
-  use std::fs::{remove_file, write};
+  use std::fs::remove_file;
   use std::io;
 
-  use dev_shared::create_temp_file;
+  use dev_shared::{create_temp_file, write_until_modified};
 
   use super::*;
 
@@ -29,7 +29,9 @@
     let stamp = stamper.stamp(&temp_file)?;
     assert_eq!(stamp, stamper.stamp(&temp_file)?);
 
-    write(&temp_file, format!("{:?}", stamp))?;
+    // Write until file modified time changes. Required on some OSs due to imprecise modified timer causing the modified
+    // stamp to be the same after fast consecutive writes.
+    write_until_modified(&temp_file, format!("{:?}", stamp))?;
     let new_stamp = stamper.stamp(&temp_file)?;
     assert_ne!(stamp, new_stamp);
     let stamp = new_stamp;
