--- pie/src/lib.rs
+++ pie/src/lib.rs
@@ -41,6 +41,14 @@
   /// Returns the default require file stamper.
   fn default_require_file_stamper(&self) -> FileStamper { FileStamper::Modified }
 
-  /// Requires given `task`, recording a dependency and selectively executing it. Returns its up-to-date output.
-  fn require_task(&mut self, task: &T) -> T::Output;
+  /// Requires given `task`, recording a dependency (using the default output stamper) and selectively executing it.
+  /// Returns its up-to-date output.
+  fn require_task(&mut self, task: &T) -> T::Output {
+    self.require_task_with_stamper(task, self.default_output_stamper())
+  }
+  /// Requires given `task`, recording a dependency (using given `stamper`) and selectively executing it. Returns its
+  /// up-to-date output.
+  fn require_task_with_stamper(&mut self, task: &T, stamper: OutputStamper) -> T::Output;
+  /// Returns the default output stamper.
+  fn default_output_stamper(&self) -> OutputStamper { OutputStamper::Equals }
 }
