--- pie/src/context/non_incremental.rs
+++ pie/src/context/non_incremental.rs
@@ -4,15 +4,16 @@
 
 use crate::{Context, Task};
 use crate::fs::open_if_file;
+use crate::stamp::{FileStamper, OutputStamper};
 
 pub struct NonIncrementalContext;
 
 impl<T: Task> Context<T> for NonIncrementalContext {
-  fn require_file<P: AsRef<Path>>(&mut self, path: P) -> Result<Option<File>, io::Error> {
+  fn require_file_with_stamper<P: AsRef<Path>>(&mut self, path: P, _stamper: FileStamper) -> Result<Option<File>, io::Error> {
     open_if_file(&path)
   }
 
-  fn require_task(&mut self, task: &T) -> T::Output {
+  fn require_task_with_stamper(&mut self, task: &T, _stamper: OutputStamper) -> T::Output {
     task.execute(self)
   }
 }
