--- pie/src/store.rs
+++ pie/src/store.rs
@@ -1,3 +1,4 @@
+use std::collections::HashMap;
 use std::path::{Path, PathBuf};
 
 use pie_graph::{DAG, Node};
@@ -9,6 +10,8 @@
 /// and query this graph.
 pub struct Store<T, O> {
   graph: DAG<NodeData<T, O>, Dependency<T, O>>,
+  file_to_node: HashMap<PathBuf, Node>,
+  task_to_node: HashMap<T, Node>,
 }
 
 #[derive(Debug)]
@@ -24,6 +27,8 @@
   fn default() -> Self {
     Self {
       graph: DAG::default(),
+      file_to_node: HashMap::default(),
+      task_to_node: HashMap::default(),
     }
   }
 }
