--- original
+++ modified
@@ -3,12 +3,11 @@
 use std::fs::{File, read_to_string, remove_file, write};
 use std::io::{self, Read, Write};
 use std::path::{Path, PathBuf};

 use dev_shared::{create_temp_dir, write_until_modified};
-use pie::{Context, Task};
-use pie::context::top_down::TopDownContext;
+use pie::{Context, Pie, Task};
 use pie::stamp::FileStamper;

 /// Enumeration over file pseudo-tasks.
 #[derive(Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Debug)]
 enum FileTask {
