--- pie/src/context/top_down.rs
+++ pie/src/context/top_down.rs
@@ -26,7 +26,11 @@
     let Some(current_executing_task_node) = &self.current_executing_task else {
       return fs::open_if_file(path); // No current executing task, so no dependency needs to be made.
     };
-    todo!()
+    let path = path.as_ref();
+    let node = self.store.get_or_create_file_node(path);
+    let (dependency, file) = FileDependency::new_with_file(path, stamper)?;
+    self.store.add_file_require_dependency(current_executing_task_node, &node, dependency);
+    Ok(file)
   }
 
   fn require_task_with_stamper(&mut self, task: &T, stamper: OutputStamper) -> T::Output {
