--- pie/src/context/top_down.rs
+++ pie/src/context/top_down.rs
@@ -10,6 +10,7 @@
 pub struct TopDownContext<T, O> {
   store: Store<T, O>,
   current_executing_task: Option<TaskNode>,
+  dependency_check_errors: Vec<io::Error>,
 }
 
 impl<T: Task> TopDownContext<T, T::Output> {
@@ -17,8 +18,13 @@
     Self {
       store: Store::default(),
       current_executing_task: None,
+      dependency_check_errors: Vec::default(),
     }
   }
+
+  pub fn get_dependency_check_errors(&self) -> impl Iterator<Item=&io::Error> {
+    self.dependency_check_errors.iter()
+  }
 }
 
 impl<T: Task> Context<T> for TopDownContext<T, T::Output> {
