--- pie/src/context/non_incremental.rs
+++ pie/src/context/non_incremental.rs
@@ -17,4 +17,27 @@
     let mut context = NonIncrementalContext;
     assert_eq!("Hello World!", context.require_task(&ReturnHelloWorld));
   }
+
+  #[test]
+  fn test_require_task() {
+    #[derive(Clone, PartialEq, Eq, Hash, Debug)]
+    enum Test {
+      ReturnHelloWorld,
+      ToLowerCase,
+    }
+
+    impl Task for Test {
+      type Output = String;
+      fn execute<C: Context<Self>>(&self, context: &mut C) -> Self::Output {
+        match self {
+          Self::ReturnHelloWorld => "Hello World!".to_string(),
+          Self::ToLowerCase => context.require_task(&Self::ReturnHelloWorld).to_lowercase(),
+        }
+      }
+    }
+
+    let mut context = NonIncrementalContext;
+    assert_eq!("Hello World!", context.require_task(&Test::ReturnHelloWorld));
+    assert_eq!("hello world!", context.require_task(&Test::ToLowerCase));
+  }
 }
